contracts/multisig-prover/src/state.rs

```diff
@@ -24,3 +28,42 @@ pub const COMMANDS_BATCH: Map<&BatchID, CommandBatch> = Map::new("command_batch"
 pub const MULTISIG_SESSION_BATCH: Map<u64, BatchID> = Map::new("multisig_session_batch");
 
 pub const REPLY_BATCH: Item<BatchID> = Item::new("reply_tracker");
+
+#[cw_serde]
+pub struct WorkerSet {
+    pub signers: Vec<Signer>,
+    pub threshold: Uint256,
+    pub id: u64, // for randomness
+}
+
+impl WorkerSet {
+    pub fn new(
+        snapshot: Snapshot,
+        pub_keys: Vec<multisig::types::PublicKey>,
+        env: Env,
+    ) -> Result<Self, ContractError> {
+        let signers = pub_keys
+            .into_iter()
+            .zip(snapshot.participants.iter())
+            .map(|(pub_key, (_, participant))| Signer {
+                address: participant.address.clone(),
+                weight: participant.weight.into(),
+                pub_key,
+            })
+            .collect();
+
+        Ok(WorkerSet {
+            signers,
+            threshold: snapshot.quorum.into(),
+            id: env.block.height,
+        })
+    }
+
+    pub fn hash(&self) -> HexBinary {
+        Keccak256::digest(serde_json::to_vec(&self).expect("couldn't serialize worker set"))
```

Comment:
Here it makes it cleaer that a BTreeSet makes way more sense than a Vec cuz order matter when it comes to the hash.